Literal floating point Narrowing Conversions

cugone opened the pull request:
Closes #155.

Last part of splitting #155 up into smaller commits.

Commit this one last.

DanRStevens commented on src/UI/Core/TextField.cpp:
```diff
@@ -139,13 +139,13 @@ void TextField::onTextInput(const std::string& _s)
 	if (mMaxCharacters > 0 && text().length() == mMaxCharacters) { return; }
 
-	int prvLen = text().length();
+	std::size_t prvLen = text().length();
```
This change appears to be included in #173. It may be more appropriate in this PR.

cugone replied:
Done

DanRStevens commented on src/States/MapViewStateDraw.cpp:
```diff
@@ -251,17 +247,15 @@ void MapViewState::drawNavInfo()
 		r.drawSubImage(mUiIcons, MOVE_DOWN_ICON.x(), MOVE_DOWN_ICON.y(), 64, 128, 32, 32, 255, 0, 0, 255);
-	}
-	else
+	} else
```
There's quite a few style changes like this, which should really be done in their own commit.

----

Not too sure how Leeor feels about the style used here. I prefer a more compact style, though I'm uncertain what the code base tends to use.

DanRStevens commented on src/States/MapViewStateDraw.cpp:
```diff
@@ -151,57 +157,47 @@ void MapViewState::drawResourceInfo()
-	if (mPinPopulationPanel	|| isPointInRect(MOUSE_COORDS.x(), MOUSE_COORDS.y(), 675, 1, 75, 19)) { mPopulationPanel.update(); }
-	if (mPinResourcePanel	|| isPointInRect(MOUSE_COORDS.x(), MOUSE_COORDS.y(), 0, 1, mResourceBreakdownPanel.width(), 19)) { mResourceBreakdownPanel.update(); }
+	if (mPinPopulationPanel || isPointInRect(MOUSE_COORDS.x(), MOUSE_COORDS.y(), 675, 1, 75, 19)) { mPopulationPanel.update(); }
+	if (mPinResourcePanel || isPointInRect(MOUSE_COORDS.x(), MOUSE_COORDS.y(), 0, 1, mResourceBreakdownPanel.width(), 19)) { mResourceBreakdownPanel.update(); }
```
More style changes that should be in their own commit.

DanRStevens commented on src/States/MapViewStateDraw.cpp:
```diff
@@ -151,57 +157,47 @@ void MapViewState::drawResourceInfo()
-	if (mCurrentMorale > mPreviousMorale) { r.drawSubImage(mUiIcons, (x + offsetX) * 10 - 17, y, 16, 48, constants::RESOURCE_ICON_SIZE, constants::RESOURCE_ICON_SIZE); }
-	else if (mCurrentMorale < mPreviousMorale) { r.drawSubImage(mUiIcons, (x + offsetX) * 10 - 17, y, 0, 48, constants::RESOURCE_ICON_SIZE, constants::RESOURCE_ICON_SIZE); }
-	else { r.drawSubImage(mUiIcons, (x + offsetX) * 10 - 17, y, 32, 48, constants::RESOURCE_ICON_SIZE, constants::RESOURCE_ICON_SIZE); }
+	if (mCurrentMorale > mPreviousMorale) { r.drawSubImage(mUiIcons, (x + offsetX) * 10 - 17, y, 16, 48, constants::RESOURCE_ICON_SIZE, constants::RESOURCE_ICON_SIZE); } else if (mCurrentMorale < mPreviousMorale) { r.drawSubImage(mUiIcons, (x + offsetX) * 10 - 17, y, 0, 48, constants::RESOURCE_ICON_SIZE, constants::RESOURCE_ICON_SIZE); } else { r.drawSubImage(mUiIcons, (x + offsetX) * 10 - 17, y, 32 ...
```
Excessively long line. Was this intentional? Other lines in this section look similar.

DanRStevens replied:
This has definitely not been resolved. The `if` and `else` blocks shouldn't all be on one long line like this.

cugone replied:
You are reviewing the wrong commit. Any change to MapViewStateDraw is part of #177. These issues have been resolved in that PR. Which is why I marked this resolved.

DanRStevens replied:
That change is on this branch. If you don't want it reviewed here, it should be removed.

DanRStevens commented on src/States/MapViewStateDraw.cpp:
```diff
@@ -69,8 +69,14 @@ void MapViewState::drawMiniMap()
+	else
+	{
+		r.drawImage(mMapDisplay, mMiniMapBoundingBox.x(), mMiniMapBoundingBox.y());
+	}
```
Reformatting change, which should be in its own commit.

DanRStevens commented on src/States/MainReportsUiState.cpp:
```diff
@@ -258,7 +257,7 @@ void MainReportsUiState::_deactivate()
 {
 	if (!active()) { return; }
-	if(key == NAS2D::EventHandler::KeyCode::KEY_ESCAPE) { exit(); }
+	if (key == NAS2D::EventHandler::KeyCode::KEY_ESCAPE) { exit(); }
```
Reformatting change.

cugone commented:
I apologize for the style and formatting changes. I was making undo commits to split up the `static_cast`s/literals and I missed a few `clang-format` auto-changes.

cugone commented:
Style and formatting changes to `MapViewStateDraw.cpp` have been moved to PR #177.

Reverted the formatting changes to `MainReportsUiState.cpp` and moved them to PR #178.

DanRStevens commented:
Thanks for looking into that. Sounds like the Clang auto formatting can be quite a pain.

There are merge conflicts now, which will need to be resolved. Might have been from the last branch I merged.

cugone commented:
Resolved.

DanRStevens commented:
I think this one needs a bit of rework and attention. There is too much going on in a single commit.

DanRStevens commented:
New merge conflicts for this branch. The code formatting still needs to be updated as well.

cugone commented:
Due to pixel-alignment requirements noted in other PRs, closing this to rework.
